crates/types/src/lib.rs

```diff
@@ -1,80 +1,78 @@
 #![recursion_limit = "256"]
-mod action;
+pub mod action;
 pub mod audio;
-mod ball;
+pub mod ball;
 pub mod ball_filter;
-mod ball_position;
-mod buttons;
+pub mod ball_position;
+pub mod buttons;
 pub mod camera_matrix;
-mod camera_position;
-mod color;
+pub mod camera_position;
+pub mod color;
 pub mod condition_input;
-mod cycle_time;
+pub mod cycle_time;
 pub mod detected_feet;
 pub mod detected_robots;
-mod fall_state;
-mod field_border;
-mod field_color;
-mod field_dimensions;
-mod field_marks;
-mod filtered_game_state;
-mod filtered_segments;
-mod filtered_whistle;
-mod game_controller_state;
-mod geometry;
+pub mod fall_state;
+pub mod field_border;
+pub mod field_color;
+pub mod field_dimensions;
+pub mod field_marks;
+pub mod filtered_game_state;
+pub mod filtered_segments;
+pub mod filtered_whistle;
+pub mod game_controller_state;
+pub mod geometry;
 pub mod grayscale_image;
 pub mod hardware;
 pub mod horizon;
-mod image_segments;
+pub mod image_segments;
 pub mod initial_look_around;
-mod initial_pose;
+pub mod initial_pose;
 pub mod interpolated;
-mod joints;
-mod joints_velocity;
-mod kick_decision;
-mod kick_step;
-mod kick_target;
-mod led;
-mod limb;
-mod line;
-mod line_data;
+pub mod joints;
+pub mod joints_velocity;
+pub mod kick_decision;
+pub mod kick_step;
+pub mod kick_target;
+pub mod led;
+pub mod limb;
+pub mod line;
+pub mod line_data;
 pub mod localization;
-mod message_event;
+pub mod message_event;
 pub mod messages;
-mod motion_command;
-mod motion_selection;
+pub mod motion_command;
+pub mod motion_selection;
 pub mod multivariate_normal_distribution;
 pub mod obstacle_filter;
-mod obstacles;
+pub mod obstacles;
 pub mod orientation_filter;
 pub mod parameters;
-mod path_obstacles;
-mod penalty_shot_direction;
-mod perspective_grid_candidates;
-mod planned_path;
-mod players;
-mod point_of_interest;
-mod primary_state;
-mod robot_dimensions;
-mod robot_kinematics;
-mod robot_masses;
-mod roles;
-mod rule_obstacles;
+pub mod path_obstacles;
+pub mod penalty_shot_direction;
+pub mod perspective_grid_candidates;
+pub mod planned_path;
+pub mod players;
+pub mod point_of_interest;
+pub mod primary_state;
+pub mod robot_dimensions;
+pub mod robot_kinematics;
+pub mod robot_masses;
+pub mod roles;
+pub mod rule_obstacles;
 pub mod samples;
-mod sensor_data;
-mod sole_pressure;
-mod sonar_obstacle;
-mod sonar_values;
-mod step_adjustment;
-mod step_plan;
-mod support_foot;
-mod walk_command;
-mod whistle;
-mod world_state;
+pub mod sensor_data;
+pub mod sole_pressure;
+pub mod sonar_obstacle;
+pub mod sonar_values;
+pub mod step_adjustment;
+pub mod step_plan;
+pub mod support_foot;
+pub mod walk_command;
+pub mod whistle;
+pub mod world_state;
 pub mod ycbcr422_image;
 
-// TODO: convert all "mod" to "pub mod"
```

Comment:
The main point of this comment was not to make all of the mods pub, but to get rid of the `pub use` section below.
At the moment the `Rgb` type can be imported as such: `use types::Rgb`. By only making all mods pub, you still allow that but also allow importing `types::color::Rgb`. The idea was to only support the latter by just exporting the namespace hierarchy instead of flattening it and exporting individual types globally.

Please remove the section below and update all `use types:...` accordingly.

Reply:
Done